packages/block-editor/src/components/block-controls/test/index.js

```diff
@@ -33,15 +39,49 @@ describe( 'BlockControls', () => {
 		},
 	];
 
-	it( 'should render a dynamic toolbar of controls', () => {
-		const wrapper = shallow(
-			<BlockEdit isSelected>
-				<BlockControls controls={ controls }>
-					<p>Child</p>
-				</BlockControls>
-			</BlockEdit>
+	beforeEach( () => {
+		const edit = ( { children } ) => <>{ children }</>;
+
+		registerBlockType( 'core/test-block', {
+			save: () => {},
+			category: 'text',
+			title: 'block title',
+			edit,
+		} );
+
+		render(
+			<SlotFillProvider>
+				<BlockEdit name="core/test-block" isSelected>
+					<BlockControls controls={ controls }>
```

Comment:
This is old behavior not used in the core anymore, but I think makes sense to test here for backward compatibility.

The current way of doing the same thing.
```js
<BlockControls>
   <ToolbarGroup controls={ controls } />
</BlockControls>
```

Reply:
Cool, thanks! I've added a test to cover that use case in 5a1c93177c8083feb38d935d8564c71b8068aa3c.